### ISSUE TITLE
Allow legend patch shapes with non-matching geometry

### ISSUE THREAD

**Issue by robintw** [qgis/QGIS#39849]
**Feature description.**
At present, when selecting the legend patch shape for a point layer, you can only select legend patches that are composed of points. This also applies when manually typing a WKT string into the text box. It would be useful in some situations to be able to select (for example) a line legend patch shape for a point layer. I can understand this may confuse some users, and so may need to be under a separate checkbox or something.

(For context, I have a point layer where at certain zooms all the points blend together into lines. It would look nicer if my legend had lines on it in that situation, rather than points).

**Comment by imanhn** [comment]
This is great suggestion and I think it is applicable to the "Symbology" part and if it could be implemented there, the Legend will be corrected as a side-effect by itself.

But here it is a problem : there is not a unique solution to the problem of connecting all the points via mutual line. This information is lies implicitly in a linear geometry by which you are telling the application, here it is a list of points, and they are in order, please connect each one to the next. With a set of random points I am afraid the feature you are asking would not be possible.

**Comment by robintw** [comment]
Yeah I'm aware that it wouldn't be possible to actually do a line symbology for a set of points without knowing which order to connect them in.

However, given that you can customise the legend patches, and they can be specified as a separate WKT string, I'd have thought it might be possible to allow legend patches to be any sort of geometry, regardless of the geometry of the actual layer.

**Comment by gioman** [comment]
Using the rule based symbology won't get you a very close thing compared to what you ask? @robintw

**Comment by robintw** [comment]
I don't know - can the rule-based symbology put lines in the legend for a point layer?

I think this is getting a bit confused: I'm not trying to change the symbology for my layer, just the legend patch that is displayed for it. Given that legend patches are already configurable, I was just asking if we can make them slightly more configurable.
